daemon/daemon_cleanup.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/cilium/cilium/daemon/defaults"
+	"github.com/cilium/cilium/pkg/bpf"
+)
+
+func runCleanup() {
+	if _, err := os.Stat(defaults.SockPath); !os.IsNotExist(err) {
+		fmt.Fprintf(os.Stderr, "Agent should not be running when cleaning up, found socket %s\n", defaults.SockPath)
+		os.Exit(1)
+	}
+
+	if !cleanupConfirmed() {
+		return
+	}
+
+	type cleanupFunc func() error
+	checks := []cleanupFunc{removeAllMaps, unmountFS, removeDirs, removeCNI}
+	for _, c := range checks {
+		if err := c(); err != nil {
+			fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+		}
+	}
+}
+
+func cleanupConfirmed() bool {
+	fmt.Printf("Desctructive operation! This will remove\n")
+	fmt.Printf("- all BPF maps\n")
+	fmt.Printf("- mounted bpffs\n")
+	fmt.Printf("- routes created by Cilium\n")
+	fmt.Printf("- library code\n")
+	fmt.Printf("- endpoint state\n")
+	fmt.Printf("- CNI configuration\n")
+	fmt.Printf("The command is non-revertable, do you want to contine [y/n]?")
+
+	var res string
+	fmt.Scanln(&res)
+	return res == "y"
+}
+
+func removeCNI() error {
+	path := "/etc/cni/net.d/10-cilium-cni.conf"
+	err := os.Remove(path)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	return err
+}
+
+func removeDirs() error {
+	dirs := []string{defaults.RuntimePath, defaults.LibraryPath}
+	for _, dir := range dirs {
+		if err := os.RemoveAll(dir); err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			return err
+		}
+	}
+
+	return nil
+}
+
+func unmountFS() error {
+	out, err := exec.Command("umount", bpf.GetMapRoot()).CombinedOutput()
+	if err == nil {//TODO: make quite on no BPF filesystem present
+		fmt.Printf("umounted bpf map root %s\n", out)
+	}
```

Comment:
`out` might be multiline (I imagine this is more likely in the error case) and that might mess up the Printf by being multiline by accident.